server/permalinks.go

```diff
@@ -98,6 +98,20 @@ func (p *Plugin) processReplacement(r replacement, glClient *gitlab.Client, wg *
 		Ref: &r.permalinkData.commit,
 	}
 	projectPath := fmt.Sprintf("%s/%s", r.permalinkData.user, r.permalinkData.repo)
+
+	// Check if project is public
+	if p.getConfiguration().EnableCodePreview != "privateAndPublic" {
```

Comment:
here this will become true for all values except `privateAndPublic`. shouldn't we check if this is equal to `public`?